### PR TITLE
[ADD] oca-gen-all-addons-tables

### PR DESCRIPTION
**Pull request by sbidoul**
Includes #41 

This script is intended to be run by the OCA bot.

### CODE REVIEW

File: tools/gen_all_addons_tables.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                call(['git', 'reset', '--hard', 'origin/' + branch], cwd=d)
+                if not os.path.isfile(os.path.join(d, 'README.md')):
+                    continue
+                call(['oca-gen-addons-table'], cwd=d,
+                     raise_fatal_error=False)
+                r = call(['git', 'diff', '--exit-code', 'README.md'], cwd=d,
+                         raise_on_error=False)
+                if r != 0:
+                    call(['git', 'commit',
+                          '-m', '[UPD] addons table in README.md', 'README.md'],
```
Comment (gurneyalex) [inline tools/gen_all_addons_tables.py]: flagged as too long by flake8

**Comment by gurneyalex** [comment]
:+1: once the small issue above is fixed

File: tools/gen_all_addons_tables.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        msg = "Command '%s' returned %d in %s" % (cmdstr, r, cwd)
+        if raise_fatal_error:
+            raise FatalError(msg)
+        else:
+            raise NonFatalError(msg)
+    return r
+
+
+def main():
+    # call(['oca-clone-everything', '--remove-old-repos'], cwd='.')
```
Comment (gurneyalex) [inline tools/gen_all_addons_tables.py]: sorry that was me signed in with the wrong account. 

==> needs fixing.

File: tools/gen_all_addons_tables.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import sys
+
+
+BRANCHES = ['8.0', '9.0']
+EXCLUDE = {
+    'maintainer-tools',
+    'maintainer-quality-tools',
+    'pylint-odoo',
+    'OCB',
+    'OpenUpgrade',
```
Comment (gurneyalex) [inline tools/gen_all_addons_tables.py]: we probably want to exclude OCA/dotnet too

Comment (sbidoul) [inline tools/gen_all_addons_tables.py]: Indeed. I'm considering to add a list of non-addons repos in oca_projects.py.

**Comment by sbidoul** [comment]
@gurneyalex all done, thanks for the review. Let me know if I can help with the implementation on the OCA server.

File: tools/gen_all_addons_tables.py (hunk 3 above)
Comment (sbidoul) [inline tools/gen_all_addons_tables.py]: Actually that filter is already in oca_project.py but buggy. #158

**Comment by moylop260** [comment]
:+1:

**Comment by gurneyalex** [comment]
:+1: (tested: cf https://github.com/OCA/stock-logistics-workflow/commit/c26ba3151b561b04c40729a2b44e6eefb2afd755)

**Comment by gurneyalex** [comment]
@sbidoul the cron is set up. Let's wait for it to trigger tonight :smile_cat:

**Comment by sbidoul** [comment]
Thanks @gurneyalex

**Pull request merged**